numpy/core/src/umath/reduction.c

```diff
@@ -493,6 +484,13 @@ PyUFunc_ReduceWrapper(PyArrayObject *operand, PyArrayObject *out,
         Py_INCREF(op_view);
     }
     else {
+        /* Cannot use where when we initialize from the operand */
```

Comment:
This prevents passing `where=np.True_` into identity-less ufuncs, despite allowing `where=True`. I think we might want to detect empty masks on a per iterator-loop basis, to allow both.

Fine to leave to another PR.

Reply:
Indeed, I prefer to leave this for another PR, as it needs some thought about what the ideal behaviour is (or even how to check whether an empty mask occurs).